app/mailers/user_notifications.rb

```diff
@@ -520,9 +528,48 @@ def send_notification_email(opts)
       show_tags_in_subject = tags.any? ? tags.join(" ") : nil
     end
 
+    group = post.topic.allowed_groups&.first
+
+    # If the group has IMAP enabled, then this will be handled by
+    # the Jobs::GroupSmtpEmail which is enqueued from the PostAlerter
+    #
+    # use_group_smtp_if_configured is used to ensure that no notifications
+    # expect for specific ones that we bless (such as user_private_message)
+    # accidentally get sent with the group SMTP settings.
+    if group.present? &&
+       group.smtp_enabled &&
+       !group.imap_enabled &&
+       SiteSetting.enable_smtp &&
+       opts[:use_group_smtp_if_configured]
+
+      port, enable_tls, enable_starttls_auto = EmailSettingsValidator.provider_specific_ssl_overrides(
+        group.smtp_server, group.smtp_port, group.smtp_ssl, group.smtp_ssl
+      )
+
+      # TODO (martin): Remove this once testing is over and this is more stable.
+      Rails.logger.warn("Using SMTP settings from group #{group.name} (#{group.id}) to send user notification for topic #{post.topic.id} and user #{user.id} (#{user.email})")
```

Comment:
mixed on merging it in with this warning, I guess if we really want this then: 

if ENV['DISCOURSE_VERBOSE_GROUP_MESSAGES']
   ...
end

Get rid of that block later on

Reply:
I don't intend to leave this in there once it goes to Meta/other places, this is mostly for my benefit when I do some testing on our hosted site. I really doubt anyone has this setting enabled right now for it to show in our logs

Reply:
OK, this is very minor, go ahead and merge as is.